wandb/sdk/internal/internal_api.py

```diff
@@ -3350,49 +3400,36 @@ def create_artifact(
                 labels: $labels,
                 aliases: $aliases,
                 metadata: $metadata,
-                {}
-                {}
-                {}
-                {}
-                {}
-            }}) {{
-                artifact {{
+                _CREATE_ARTIFACT_ADDITIONAL_VALUE_
+            }) {
+                artifact {
                     id
                     digest
                     state
-                    aliases {{
+                    aliases {
                         artifactCollectionName
                         alias
-                    }}
-                    artifactSequence {{
+                    }
+                    artifactSequence {
                         id
-                        latestArtifact {{
+                        latestArtifact {
                             id
                             versionIndex
-                        }}
-                    }}
-                }}
-            }}
-        }}
-        """.format(
-                "$historyStep: Int64!,"
-                if can_handle_history and history_step not in [0, None]
-                else "",
-                "$distributedID: String," if distributed_id else "",
-                "$clientID: ID!," if can_handle_client_id else "",
-                "$sequenceClientID: ID!," if can_handle_client_id else "",
-                "$enableDigestDeduplication: Boolean," if can_handle_dedupe else "",
-                # line sep
-                "historyStep: $historyStep,"
-                if can_handle_history and history_step not in [0, None]
-                else "",
-                "distributedID: $distributedID," if distributed_id else "",
-                "clientID: $clientID," if can_handle_client_id else "",
-                "sequenceClientID: $sequenceClientID," if can_handle_client_id else "",
-                "enableDigestDeduplication: $enableDigestDeduplication,"
-                if can_handle_dedupe
-                else "",
+                        }
+                    }
+                }
+            }
+        }
+        """
+
+        fields = self.server_create_artifact_introspection()
+        if "ttlDurationSeconds" not in fields:
+            logger.error(
```

Comment:
Nit: this should be a warning not an error, since it doesn't stop the artifact creation. Also, let's only show it if someone actually tries to set a TTL.